vcf.c

```diff
@@ -2338,9 +2337,26 @@ static int vcf_parse_format(kstring_t *s, const bcf_hdr_t *h, bcf1_t *v, char *p
             v->errcode |= BCF_ERR_TAG_INVALID;
             return -1;
         }
-        align_mem(mem);
+        if (align_mem(mem) < 0) {
+            hts_log_error("Memory allocation failure");
+            v->errcode |= BCF_ERR_LIMITS;
+            return -1;
+        }
         f->offset = mem->l;
-        ks_resize(mem, mem->l + v->n_sample * f->size);
+
+        // Limit the total memory to ~2Gb per VCF row.  This should mean
+        // malformed VCF data is less likely to take excessive memory and/or
+        // time.
+        if (v->n_sample * (size_t)f->size > INT_MAX) {
```

Comment:
The cast needs to be to `uint64_t`, as with `size_t` the multiplication could still wrap around on 32-bit platforms.  For the same reason, if this limitation is ever relaxed the test would have to be kept but with `INT_MAX` changed to `SIZE_MAX`.